doc/drop_counters/drop_counters_HLD.md

```diff
@@ -0,0 +1,268 @@
+| Drop Category                | SAI Counter                                  | Configurable? |
+|------------------------------|----------------------------------------------|:-------------:|
+| RX Layer-2 packet corruption | SAI_PORT_STAT_IF_IN_ERRORS                   | No            |
+| TX Layer-2 packet corruption | SAI_PORT_STAT_IF_OUT_ERRORS                  | No            |
+| RX MMU packet drop           | SAI_PORT_STAT_IF_IN_DROPPED_PKTS             | No            |
+| TX MMU packet drop           | SAI_PORT_STAT_IF_OUT_DROPPED_PKTS            | No            |
+| All RX pipeline drops        | SAI_PORT_STAT_IF_IN_DISCARDS                 | No            |
+| All TX pipeline drops        | SAI_PORT_STAT_IF_OUT_DISCARDS                | No            |
+| Specified RX pipeline drops  | SAI_DEBUG_COUNTER_TYPE_PORT_IN_DROP_REASONS  | Yes           |
+| Specified TX pipeline drops  | SAI_DEBUG_COUNTER_TYPE_PORT_OUT_DROP_REASONS | Yes           |
+
+# 2 Requirements
+
+## 2.1 Functional Requirements
+1. Users can configure debug counters to track one or more drop reason(s).
+    1. Supported ingress drop reasons are specified in sai_port_in_drop_reason_t.
+    2. Supported egress drop reasons are specified in sai_port_out_drop_reason_t.
+2. Users can access debug counter information via a CLI command
+    1. Users can see what types of drops each counter contains
+    2. Users can assign aliases to counters
+    3. Users can clear counters
+
+## 2.2 Configuration and Management Requirements
+Configuration of the drop counters can be done via:
+* CLI
+* JSON input
+
+## 2.3 Scalability Requirements
+Users must be able to use all counters and drop reasons provided by the underlying hardware.
+
+# 3 Design
+
+## 3.1 Counters DB
+The contents of the drop counters will be added to Counters DB by flex counters.
+
+## 3.2 Config DB
+We'll add two new tables to Config DB:
+* DEBUG_COUNTER to track which counters have been configured and for what purpose
+    * At this point the only supported type is PACKET_DROP
+* PACKET_DROP_COUNTER to save drop counters that have been configured by the user
+
+### 3.2.1 DEBUG_COUNTER Table
+Example:
+```
+{
+    "DEBUG_COUNTER": {
+        "DEBUG_0": {
+            "configured": true,
+            "type": "PACKET_DROP"
+        },
+        "DEBUG_1": {
+            "configured": true,
+            "type": "PACKET_DROP"
+        },
+        "DEBUG_2": {
+            "configured": true,
+            "type": "PACKET_DROP"
+        },
+        "DEBUG_3": {
+            "configured": false
+        }
+    }
+}
+```
+
+### 3.2.2 PACKET_DROP_COUNTER Table
+Example:
+```
+{
+    "PACKET_DROP_COUNTER": {
+        "LEGAL_RX_DROPS": {
+            "counter": "DEBUG_0",
+            "type": "ingress",
+            "reasons": [
+                SMAC_EQUALS_DMAC,
+                INGRESS_VLAN_FILTER
+            ],
+            "desc": "Legal RX pipeline drops"
+        },
+        "LEGAL_TX_DROPS": {
```

Comment:
`LEGAL `or `LEGIT `is a better name for this?

Reply:
I don't think it makes too much of a difference since this is just an example, but I can update it if people feel like a different name would be more clear.